refinery/data_set_manager/utils.py

```diff
@@ -635,91 +631,108 @@ def generate_solr_params(
         facets_from_config=False,
         exclude_facets=[]):
     """
-    Either returns a solr url parameter string,
+    Either returns a solr parameters obj,
     or None if assay_uuids is empty.
     """
-
+    facet_field = params.get('facets')
+    facet_filter = params.get('filter_attribute')
     is_annotation = params.get('is_annotation', 'false')
-    facet_count = params.get('include_facet_count', 'true')
-    start = params.get('offset', '0')
     # row number suggested by solr docs, since there's no unlimited option
     row = params.get('limit', str(constants.REFINERY_SOLR_DOC_LIMIT))
-    field_limit = params.get('attributes')
-    facet_field = params.get('facets')
-    facet_pivot = params.get('pivots')
+    start = params.get('offset', '0')
     sort = params.get('sort')
-    facet_filter = params.get('filter_attribute')
 
-    fixed_solr_params = \
-        '&'.join(['fq=is_annotation:%s' % is_annotation,
-                  'start=%s' % start,
-                  'rows=%s' % row,
-                  'q=django_ct:data_set_manager.node&wt=json',
-                  'facet=%s' % facet_count,
-                  'facet.limit=-1'
-                  ])
+    fixed_solr_params = {
+        "facet.limit": "-1",
+        "fq": "is_annotation:{}".format(is_annotation),
+        "rows": row,
+        "start": start,
+        "wt": "json"
+    }
 
     if len(assay_uuids) == 0:
         return None
-    solr_params = 'fq=assay_uuid:({})'.format(' OR '.join(assay_uuids))
-
-    fq = params.get('fq')
-    if fq is not None:
-        solr_params += '&fq=' + fq
+    filter_arr = ['assay_uuid:({})'.format(' OR '.join(assay_uuids))]
 
+    field_limit = []  # limit attributes to return
+    facet_fields_obj = {}  # requested facets formatted for solr
     if facets_from_config:
+        facet_field = ''
         # Twice as many facets as necessary, but easier than the alternative.
-        facet_template = '{0}_Characteristics{1},{0}_Factor_Value{1}'
-        facet_field = ','.join(
-            [facet_template.format(s, NodeIndex.GENERIC_SUFFIX) for s
-             in settings.USER_FILES_FACETS.split(",")]
-        )
-        solr_params += '&fl=*{},name,*_uuid,type,django_id,{}'.format(
-            NodeIndex.GENERIC_SUFFIX, NodeIndex.DOWNLOAD_URL)
+        for facet in settings.USER_FILES_FACETS.split(","):
+            char_str = '{}_Characteristics{}'.format(
+                    facet,
+                    NodeIndex.GENERIC_SUFFIX
+                )
+            factor_str = '{}_Factor_Value{}'.format(
+                    facet,
+                    NodeIndex.GENERIC_SUFFIX
+                )
+
+            if not facet_field:
+                facet_field = ','.join([char_str, factor_str])
+            else:
+                facet_field = ','.join([facet_field, char_str, factor_str])
```

Comment:
Maybe build up a list and join it once at the end instead?

Reply:
I was doing other things here before, I'll just revert to the previous code now.